main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -811,6 +814,18 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
       }
     }
 
+    def mkZipFile(zipFile: Path): Validation[(ZipOutputStream => Unit) => Validation[Unit, BootstrapError], BootstrapError] = {
+      Files.createDirectories(zipFile.getParent.normalize())
+      val zip: (ZipOutputStream => Unit) => Try[Unit] = Using(new ZipOutputStream(Files.newOutputStream(zipFile)))
+      val callback = (contents: ZipOutputStream => Unit) => {
+        zip(contents) match {
+          case Success(()) => Validation.Success(())
+          case Failure(exception) => Validation.Failure(BootstrapError.FileError(exception.getMessage))
+        }
+      }
+      Validation.Success(callback)
+    }
+
```

Comment:
Should this return a validation or just a callback? I.e., `Validation[(ZipOutputStream => Unit) => Validation[Unit, BootstrapError], BootstrapError]` or just `(ZipOutputStream => Unit) => Validation[Unit, BootstrapError]` since the result is always `Validation.Success`?

Reply:
Why does it need such a complex type signature?

Reply:
It wraps the `Using` type in a `Validation` so it can be used in a flatmap. It partially applies `Using` which returns a function that gives access to the zip file. However, the result of that function is a `Try` which is then mapped the a `Validation.

Reply:
`mkZipFile` could also mirror the interface of `Using`:
```scala
def mkZipFile(file: => ZipOutputStream)(addTo: ZipOutputStream => Unit): Validation[Unit, BootstrapError]
```

Reply:
And then just call `Using` directly at the use site?